### PR TITLE
Keyboard typo

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation

implement `langcheck.augment.keyboard_typo()`

### CODE REVIEW

File: src/langcheck/augment/en/_keyboard_typo.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from typing import List
+
+import nlpaug.augmenter.char as nac
+
+
+def keyboard_typo(
+    texts: List[str],
```
Comment (liwii) [inline src/langcheck/augment/en/_keyboard_typo.py:7]: Could you also make it accept single `str`? (Arguments for metrics are so updated recently).

File: src/langcheck/augment/en/_keyboard_typo.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    **kwargs,
+) -> List[str]:
+    '''Generate keyboard typo perturbed texts for augmentation.
+
+    Args:
+        texts: List of texts to be augmented.
```
Comment (liwii) [inline src/langcheck/augment/en/_keyboard_typo.py:13]: I think showing some arguments of `nlpaug.augmenter.char.keyboard.KeyboardAug` explicitly here is helpful for langcheck users. At least

- aug_char_p (float)
- aug_char_min (int)
- aug_char_max (int)
- aug_word_p (float)
- aug_word_min (int)
- aug_word_max (int)

seem general enough to add to the list of langcheck arguments. WDYT?

File: src/langcheck/augment/en/_keyboard_typo.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        **kwargs: Arbitrary keyword arguments, which will be passed to
+            :class:`nlpaug.augmenter.char.keyboard.KeyboardAug`. See the
+            documentation for more details.
```
Comment (liwii) [inline src/langcheck/augment/en/_keyboard_typo.py:16]: Could you add the link to their readthedocs?
https://nlpaug.readthedocs.io/en/latest/augmenter/char/keyboard.html

File: src/langcheck/augment/en/_keyboard_typo.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+    Returns:
+        A list of perturbed texts.
+    '''
+    aug = nac.KeyboardAug()
+    return aug.augment(texts, **kwargs)
```
Comment (liwii) [inline src/langcheck/augment/en/_keyboard_typo.py:22]: Seemingly these `kwargs` should be passed to `nac.KeyboardAug()` constructor?

File: tests/augment/test_keboard_typo.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+import pytest
+
+from langcheck.augment import keyboard_typo
+
+
+@pytest.mark.parametrize(
+    "texts, expected",
+    [
+        (["Hello, world!"], ["HF;lo, eorlE!"]),
```
Comment (liwii) [inline tests/augment/test_keboard_typo.py:12]: The perturbation seems too much as the default setting to me. How about using some values like `aug_char_p=0.1` as the default?

```python
>>> import nlpaug.augmenter.char as nac
>>> import random
>>> random.seed(42)
>>> nac.KeyboardAug().augment(['Hello, world!!'])
['HF;lo, eorlE!!']
>>> nac.KeyboardAug(aug_char_p=0.1).augment(['Hello, world!!'])
['He?lo, w*rld!!']
```

Comment (liwii) [reply tests/augment/test_keboard_typo.py:12]: `include_special_char=False, include_numeric=False` should be reasonable as the default setting too

**Review by liwii — CHANGES_REQUESTED** [review]
Thanks for quick implementation! Left a few comments.

**Comment by Alnusjaponica** [comment]
@liwii I have resolved all the comments. PTAL.

File: src/langcheck/augment/__init__.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+from langcheck.augment import en
+
+__all__ = [
+    "en",
```
Comment (liwii) [inline src/langcheck/augment/__init__.py:4]: From `langcheck.metrics` you can also directly call English metrics without `en` module. Could you also do that in `augment` for consistency? (I do not think it is a very good design, but let's re-design them altogether later)

File: src/langcheck/augment/en/_keyboard_typo.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from nlpaug.augmenter.char.keyboard import KeyboardAug
+
+
+def keyboard_typo(
+    texts: list[str] | str,
```
Comment (liwii) [inline src/langcheck/augment/en/_keyboard_typo.py:7]:
```suggestion
    texts: List[str] | str,
```
For consistency with other parts of the codebase?

File: src/langcheck/augment/en/_keyboard_typo.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    **kwargs,
+) -> list[str]:
```
Comment (liwii) [inline src/langcheck/augment/en/_keyboard_typo.py:9]:
```suggestion
) -> List[str]:
```

File: src/langcheck/augment/en/_keyboard_typo.py [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        listed below:
+            - ``aug_char_p`` (float): Percentage of character (per token)
+              will be augmented.
+            - ``aug_char_min`` (int): Minimum number of character will be
+              augmented.
+            - ``aug_char_max`` (int): Maximum number of character will be
+              augmented.
+            - ``aug_word_p`` (float): Percentage of word will be augmented.
+            - ``aug_word_min`` (int): Minimum number of word will be augmented.
+            - ``aug_word_max`` (int): Maximum number of word will be augmented.
```
Comment (liwii) [inline src/langcheck/augment/en/_keyboard_typo.py:29]:
```suggestion

        - ``aug_char_p`` (float): Percentage of character (per token)
          will be augmented.
        - ``aug_char_min`` (int): Minimum number of character will be
          augmented.
        - ``aug_char_max`` (int): Maximum number of character will be
          augmented.
        - ``aug_word_p`` (float): Percentage of word will be augmented.
        - ``aug_word_min`` (int): Minimum number of word will be augmented.
        - ``aug_word_max`` (int): Maximum number of word will be augmented.

```

There are some errors when building the doc. Seemingly unindenting the bullet list & inserting blanks before & after that will resolve the issue.

**Review by liwii — APPROVED** [review]
LGTM after resolving the last few points!!

File: src/langcheck/augment/en/_keyboard_typo.py (hunk 7 above)
Comment (Alnusjaponica) [reply src/langcheck/augment/en/_keyboard_typo.py:7]: Using `typing.List` is deprecated by [PEP585](https://peps.python.org/pep-0585/#:~:text=Importing%20those%20from%20typing%20is%20deprecated.). I'll create an issue to handle them but should I restore these change for now?

Comment (liwii) [reply src/langcheck/augment/en/_keyboard_typo.py:7]: Ah good to here, then let's leave this line unchanged. Thanks for creating the issue for that!

File: src/langcheck/augment/__init__.py (hunk 6 above)
Comment (Alnusjaponica) [reply src/langcheck/augment/__init__.py]: Import path updated

File: src/langcheck/augment/en/_keyboard_typo.py (hunk 9 above)
Comment (Alnusjaponica) [reply src/langcheck/augment/en/_keyboard_typo.py]: I fixed the sphinx error.

**Comment by liwii** [comment]
Did a final review and LGTM! Feel free to merge, thanks for the changes!!

**Pull request merged**